packages/at_onboarding_flutter/lib/src/screen/at_onboarding_home_screen.dart

```diff
@@ -710,174 +648,81 @@ class _AtOnboardingHomeScreenState extends State<AtOnboardingHomeScreen> {
-            setState(() {
-              loading = false;
-            });
-            return;
-          }
-        }
-      }
-      if (aesKey == null && atsign == null && fileContents != null) {
-        List<String> keyData = fileContents.split(',"@');
-        List<String> params = keyData[1].toString().substring(0, keyData[1].length - 2).split('":"');
-        atsign = "@${params[0]}";
-        Map<String, dynamic> keyMap = jsonDecode(fileContents);
-        aesKey = keyMap[AtOnboardingConstants.atSelfEncryptionKey];
-      }
-      if (fileContents == null || (aesKey == null && atsign == null)) {
-        await showErrorDialog(_incorrectKeyFile);
-        setState(() {
-          loading = false;
-        });
-        return;
-      } else if (OnboardingService.getInstance().formatAtSign(atsign) != _pairingAtsign && _pairingAtsign != null) {
-        await showErrorDialog(AtOnboardingErrorToString().atsignMismatch(_pairingAtsign));
-        setState(() {
-          loading = false;
-        });
-        return;
-      }
-      setState(() {
-        loading = false;
-      });
-      await _processAESKey(atsign, aesKey, fileContents);
-    } catch (error) {
-      setState(() {
-        loading = false;
-      });
-      _logger.severe('Uploading backup zip file throws $error');
-      await showErrorDialog(_failedFileProcessing);
     }
   }
 
-  Future<void> _uploadKeyFileForDesktop() async {
-    try {
-      _isContinue = true;
-      String? fileContents, aesKey, atsign;
-      setState(() {
-        loading = true;
-      });
-
-      String? path = await _desktopKeyPicker();
-      if (path == null) {
-        setState(() {
-          loading = false;
-        });
-        return;
-      }
-
-      File selectedFile = File(path);
-      int length = selectedFile.lengthSync();
-      if (length < 10) {
-        await showErrorDialog(_incorrectKeyFile);
-        return;
-      }
-
-      fileContents = File(path).readAsStringSync();
-      // ignore: unnecessary_null_comparison
-      if (aesKey == null && atsign == null && fileContents.isNotEmpty) {
-        List<String> keyData = fileContents.split(',"@');
-        List<String> params = keyData[1].toString().substring(0, keyData[1].length - 2).split('":"');
-        atsign = "@${params[0]}";
-        Map<String, dynamic> keyMap = jsonDecode(fileContents);
-        aesKey = keyMap[AtOnboardingConstants.atSelfEncryptionKey];
-      }
-      if (fileContents.isEmpty || (aesKey == null && atsign == null)) {
-        await showErrorDialog(_incorrectKeyFile);
-        setState(() {
-          loading = false;
-        });
-        return;
-      } else if (OnboardingService.getInstance().formatAtSign(atsign) != _pairingAtsign && _pairingAtsign != null) {
-        await showErrorDialog(AtOnboardingErrorToString().atsignMismatch(_pairingAtsign));
-        setState(() {
-          loading = false;
-        });
-        return;
+  Future<void> _uploadKeyFile() async {
```

Comment:
This calls the new service, and listens to the progress stream, updating the UI accordingly. The actual logic behind when UI changes occur is the same